src/app/treelist-checkbox/treelist-checkbox.component.ts

```diff
@@ -0,0 +1,250 @@
+import { AfterViewInit, Component, EventEmitter, Input, OnInit, Output, ViewChild } from '@angular/core';
+import { DatasetNode } from 'app/dataset-node/dataset-node';
+@Component({
+  selector: 'gpf-treelist-checkbox',
+  templateUrl: './treelist-checkbox.component.html',
+  styleUrls: ['./treelist-checkbox.component.css']
+})
+export class TreelistCheckboxComponent implements OnInit, AfterViewInit {
+  public toggleSwitch(event): void {
+    if (!event.target.value) {
+      event.target.value = 'true';
+      return;
+    }
+    if (event.target.value === 'true') {
+      event.target.value = 'false';
+    } else {
+      event.target.value = 'true';
+    }
+  }
+  @Output() public changeCheckStatusEvent = new EventEmitter<string>();
+
+  @Input()
+  public data: DatasetNode;
+
+  @Input()
+  public allStudies = new Set<string>();
+
+  @Input()
+  private fillStatesEvent = new EventEmitter<Set<string>>();
+
+  @ViewChild('hide') public input;
+
+  public ngOnInit(): void {
+    this.allStudies.delete(this.data.dataset.name);
+    this.allStudies.forEach(study => {
+      this.allStudies.delete(study);
+    });
+  }
+
+  public ngAfterViewInit(): void {
+    this.fillStatesEvent.subscribe(selected => {
+      this.allStudies = selected;
+      this.updateFilterStates(this.data);
+    });
+  }
+
+  public updateFilters($event): void {
+    const datasetNode = this.findNodeById(this.data, $event.target.id);
+    if ($event.target.checked) {
+      if (datasetNode.children.length > 0) {
+        this.getAllChildren(datasetNode.children, new Set(), true).forEach(child => {
+          this.allStudies.add(child);
+        });
+        this.allStudies.add(datasetNode.dataset.name);
+      } else {
+        this.allStudies.add(datasetNode.dataset.name);
+        this.updateParents(datasetNode);
+      }
+    } else if (datasetNode?.children.length > 0) {
+      this.getAllChildren(datasetNode.children, new Set(), true).forEach(child => {
+        this.changeCheckStatusEvent.emit($event.target.id);
+        this.allStudies.delete(child);
+      });
+      this.allStudies.delete($event.target.id);
+    } else if (this.allStudies.has($event.target.id)) {
+      this.changeCheckStatusEvent.emit($event.target.id);
+      this.allStudies.delete($event.target.id);
+    }
+    if (this.allStudies.has($event.target.id)) {
+      this.changeCheckStatusEvent.emit($event.target.id);
+      this.allStudies.add($event.target.id);
+    } else {
+      this.changeCheckStatusEvent.emit($event.target.id);
+      this.allStudies.delete($event.target.id);
+    }
+    this.updateFilterStates(this.data);
+  }
```

Comment:
Lots of repeating calls that can be found in both the `if` and `else` cases. You should extract them outside the if/else blocks.